libqtile/dgroups.py

```diff
@@ -64,6 +65,19 @@ def func(dgroup):
 
     return func
 
+def _match_ppid(pid: int):
+    """Utility function to match pid againts window.pid or any of its ppid"""
+    
+    def func(w: Window) -> bool:
+        cid = w.get_pid()
+        while cid not in (pid, 0, 1): # if 0 or 1 are reached, we cannot really find parents that are not qtile
```

Comment:
Can we check the cmdline here or something too? This seems like it could accidentally be used to match other windows if e.g. i spawn something from alacritty or whatever.

Reply:
> I dont think checking the cmdline would work properly all the times either (name in /stat file generally is the same, but there's no guarantee either).

Sure, it's all heuristics at the end of the day. If it's default-off then maybe that's ok along with a note that it is potentially too aggressive when things are spawned from a terminal.

Reply:
> > I dont think checking the cmdline would work properly all the times either (name in /stat file generally is the same, but there's no guarantee either).
> 
> Sure, it's all heuristics at the end of the day. If it's default-off then maybe that's ok along with a note that it is potentially too aggressive when things are spawned from a terminal.

I'll work on a solution without psutils (as suggested above) and with more
straightforward documentation on "don't use this unless needed"